context-free.md

```diff
@@ -1,22 +1,26 @@
 # Context Free
 
 ##  Who is this programming language for?
+The language is for people with an interest and affinity for art, programming, and grammars.  It provides a convenient combination of each of those things, allowing anyone with a sense for at least one of them to pick it up rather easily.
 
 
 ## What is easy to do in this language? Why is it easy?
+It is very easy to define something recursively, by simply defining your grammar and then letting the rendering engine take over.  It's easy because it requires very little manual entry by the artist.  Simple shapes and variations of simple shapes are also easy.
 
 
 ## What is hard to do in this language? Why is it hard?
+The non-recursive sections of the image were hard - it required a lot more manual fiddling.  Additionally, I had a really hard time with overlapping sections as well as shape parameters (which I gave up on) and loops (which I mostly replaced with recursion).  
 
 
 ## How did you learn how to program in this language?
+I started with their welcome program, and then changed incrementally using their documentation, trial and error, and the submissions of my peers.
 
 
 ## What is the underlying _computational model_ for this programming language? 
-_We don't yet have a great definition of the term "computational model". 
-For now, try to come up with the clearest, most concise explanation of what 
-happens when a ContextFree program runs._
+It finds the first shape, defined by ``startshape``, and then attempts to build that shape using the other defined rules.  It treats these like a Context Free Grammar and non-deterministically creates an image. 
 
 
 ## What do you think is interesting about the ContextFree program you wrote?
+I like my knight.  He doesn't have any legs, but he seems tough.
 
+I also like his rainbow arsenal.  I think they give him some real panache.
```

Comment:
Anything computationally interesting? Maybe reflect on why it was difficult/resulted in a long program. What sort of design does the language encourage or discourage? Why?